exp/views/responses.py

```diff
@@ -1109,6 +1165,78 @@ def get(self, request, *args, **kwargs):
         return response
 
 
+class StudyResponsesFrameDataPsychDS(ResponseDownloadMixin, generic.list.ListView):
+    """Hitting this URL downloads a ZIP file in Psych-DS formatting with frame data from one response per file in CSV format"""
+
+    # TODO: with large files / many responses generation can take a while. Should generate asynchronously along
+    # with the data dict.
+    def render_to_response(self, context, **response_kwargs):
+        paginator = context["paginator"]
+        study = self.study
+
+        if study.study_type.is_external:
+            messages.error(
+                self.request, "Frame data is not available for External Studies."
+            )
+
+            return study_responses_all(study)
+
+        # build dict with contextual infomation about the study/lab/user
+        metadata_json = build_metadata_object(study)
+        # variables from column headers will be collected while looping through files
+        variables_measured = []
+
+        zipped_file = io.BytesIO()
+        with zipfile.ZipFile(zipped_file, "w", zipfile.ZIP_DEFLATED) as zipped:
+            for page_num in paginator.page_range:
+                page_of_responses = paginator.page(page_num)
+                for resp in page_of_responses:
+                    data = build_single_response_framedata_csv(resp)
+                    # get column headers from first row
+                    variables_measured += [
+                        column.strip('"')
+                        for column in data.split("\n")[0].strip().split(",")
+                    ]
+                    # psych-DS files use "keyword" formatting
+                    keywords = {
+                        "study": study_name_for_files(study.name),
+                        "response": resp.id,
+                        "child": resp.child_id,
+                    }
+                    filename = keyword_filename(keywords, "csv")
+                    zipped.writestr(f"/data/{filename}", data)
+                    # make a response-specific metadata file
+                    sidecar_metadata = {
+                        "response_uuid": str(resp.uuid),
+                        "eligibility": resp.eligibility[0],
+                        "study_completed": resp.completed,
+                    }
+                    sidecar_filename = keyword_filename(keywords, "json")
+                    zipped.writestr(
+                        f"/data/{sidecar_filename}",
+                        json.dumps(sidecar_metadata, indent=4),
+                    )
+            zipped.writestr(
+                "/materials/study_protocol.json", json.dumps(study.structure, indent=4)
+            )
```

Comment:
For studies that use the protocol generator, would it make sense to include the JavaScript (whether formatted or not)? I tried downloading a study that uses the protocol generator (JS) instead of the JSON protocol, and I got this study protocol file with:
```json
{
    "exact_text": "{\"frames\": {}, \"sequence\": []}"
}
```
I'm wondering if it would make sense to include the protocol generate JS (instead of or in addition to this study_protocol file)? Or maybe just note it in the study metadata?